keboola/component_configuration_resource.go

```diff
@@ -0,0 +1,426 @@
+}
+
+// Metadata returns the resource type name.
+func (r *configResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_component_configuration"
+}
+
+// GetSchema defines the schema for the resource.
+func (r *configResource) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: "Manages component configurations (https://keboola.docs.apiary.io/#reference/components-and-configurations).",
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Description: "Unique string identifier assembled as branchId/componentId/configId.",
+				Type:        types.StringType,
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					resource.UseStateForUnknown(),
+				},
+			},
+			"configuration_id": {
+				Description: "Id of the configuration. If not specified, then will be autogenerated.",
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+			},
+			"component_id": {
+				Description: "Id of the component.",
+				Type:        types.StringType,
+				Required:    true,
+			},
+			"branch_id": {
+				Description: "Id of the branch. If not specified, then default branch will be used.",
+				Type:        types.Int64Type,
+				Optional:    true,
+				Computed:    true,
+			},
+			"name": {
+				Description: "Name of the configuration.",
+				Type:        types.StringType,
+				Required:    true,
+			},
+			"description": {
+				Description: "Description of the configuration.",
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+			},
+			"change_description": {
+				Description: "Change description associated with the configuration change.",
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+			},
+			"is_disabled": {
+				Description: "Wheter configuration is enabled or disabled.",
+				Type:        types.BoolType,
+				Optional:    true,
+				Computed:    true,
+			},
+			"configuration": {
+				Description: "Content of the configuration specified as JSON string.",
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+			},
+			"is_deleted": {
+				Description: "Wheter configuration has been deleted or not.",
+				Type:        types.BoolType,
+				Computed:    true,
+			},
+			"created": {
+				Description: "Timestamp of the configuration creation date.",
+				Type:        types.StringType,
+				Computed:    true,
+			},
+			"version": {
+				Description: "Id of the version",
+				Type:        types.Int64Type,
+				Computed:    true,
+			},
+		},
+	}, nil
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *configResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Retrieve values from plan
+	var plan configModel
+	diags := req.Plan.Get(ctx, &plan)
```

Comment:
pri create sa na zaciatku nacita Plan - to vlastne vrati to je definovane v tom terraform resource. Nasledne to posle na kbc api a ulozi do state. Vid tutorial https://developer.hashicorp.com/terraform/tutorials/providers-plugin-framework/providers-plugin-framework-resource-create